### DISCUSSION TITLE
feat(datasource/custom): allow sending custom headers

### DISCUSSION THREAD

**Discussion by weikinhuang** [renovatebot/renovate#25677] [Suggest an Idea]
### Tell us more.

**Describe the proposed change(s).**

Certain endpoints require special auth header to access the correct content. An example, and my use case is to retrieve the plex version. Without a `X-Plex-Token` header, plex's api `https://plex.tv/api/downloads/5.json` returns only stable versions. With the header the api returns beta versions instead (which is the behavior I would like).

**Comment by rarkins** [comment]
It's possible we might want to make this a generic `hostRules` option for any datasource instead of custom-specific. @secustor wdyt?

**Reply by weikinhuang** [reply]
That would be even more useful, as right now host rules are very limited in the types of extra data that can be sent. If host rules were used, it would be good to be able to populate the value from env vars as well.

**Reply by secustor** [reply]
Both is possible, but host rules would be more useful IMO as we could support more complex routing setups.

**Reply by rarkins** [reply]
@weikinhuang all config can be populated via env one way or another

**Reply by viceice** [reply]
I like the idea to configure headers via hostrule

**Reply by secustor** [reply]
https://github.com/renovatebot/renovate/issues/26625

**Comment by stuart-warren** [comment]
We'd like this for being able to access the snapcraft api

https://api.snapcraft.io/docs/info.html

**Comment by viceice** [comment]
- https://github.com/renovatebot/renovate/pull/26225
